### DISCUSSION TITLE
Invalid websocket upgrade

### DISCUSSION THREAD

**Discussion by Chenger1** [miguelgrinberg/python-socketio#1112] [Q&A]
script
```python
import logging
import socketio
from aiohttp import web
from settings import Config

sio = socketio.AsyncServer(async_mode='aiohttp',
                           cors_allowed_origins="*",
                           engineio_logger=True,
                           logger=True)

app = web.Application(debug=Config.DEBUG)
sio.attach(app)

@sio.event
async def connect(sid, environ, *args, **kwargs):
    print(sid)
```

nginx
```nginx
server {
    listen 8003;
    server_name 0.0.0.0;
    client_max_body_size 100M;

    location / {
        proxy_pass http://socketio;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $host;
        proxy_redirect off;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
    }

    location /media/ {
        alias /usr/src/app/media/;
    }
}
```
**This problem is shown after move socket.io to https under nginx.**
**Before i ran in independent docker container without nginx configuration.**
If i try to connect via postman (socket.io request) - i get "Error: Unexpected server response: 400". 
In terminal i have only this: 
`INFO:engineio.server:Invalid websocket upgrade`
It seems like postman doesnt send Upgrade header, but it does:
```
Request Headers
Sec-WebSocket-Version: 13
Sec-WebSocket-Key: key
Connection: Upgrade
Upgrade: websocket
Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits
Host: myhost.com
```

Has someone faced this problem before?

**Comment by miguelgrinberg** [comment]
Hard to know what the problem is, but the error does mean that the WebSocket connection request is missing something. You may need to hand edit the server.py file of the python-engineio package which issues this error to log the request at the time the error is printed. That would be a relatively easy way to figure out what is missing.

**Reply by Chenger1** [reply]
I found that nginx missed "Upgrade" header, maybe because of another central server configuration.
But if i set:
`nginx
proxy_set_header Upgrade websocket;
`
i cant connect. Engineio logged:
`
Received request to upgrade to websocket.
Upgrade to websocket successful
`
and thats all.

**Reply by miguelgrinberg** [reply]
Forcing the `Upgrade` header does not seem like a good solution, as that would prevent the initial long-polling connections from working. All the requests are going to look like WebSocket upgrades, which is not what you want.
